routes/items.js

```diff
@@ -8,24 +8,37 @@ router.get("/", (req, res, next) => {
 });
 
 router.get("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
-  res.status(501).send({ error: 'route not yet implemented' });
+  // TODO: complete writing this route handler  
+  let item = itemData.getById(req.params.id); //req.params.id is a path param needed since it is a dynmaic id in the path 
+  if(item){
+    res.status(200).send(item);
+  } else {
+    res.status(404).send({ error: 'item does not exist' });
+  }
 });
 
 router.post("/", (req, res, next) => {
-  itemData.create(req.body);
+  itemData.create(req.body); //req.body is the argument needed here so it can update the body of the item
   res.sendStatus(200);
 });
 
 router.put("/:id", (req, res, next) => {
-  // TODO: complete writing this route handler
+  let item =  itemData.updateById(req.params.id, req.body); //needs 2 args, params.id to get the id on the dynmaic path and req.body to update the bidy
```

Comment:
I like this logical check, but it won't work as expected right now. `itemData.updateById` method doesn't return anything, so `item` will always be null. If that method were refactored to return the updated item, then this would work.